### PR TITLE
Fix lexer and tokenizer to retain line breaks properly

### LINKED ISSUE #2340: The lexer appears to handle new lines incorrectly
**Marked version:**
4.0.6

**Describe the bug**
When using the lexer, it ignores new lines under some circumstances. 

**To Reproduce**
```
console.log(marked.lexer("\nT\nh\n---"))
```
returns 
```json
[
    {
        "type": "paragraph",
        "raw": "T\nh",
        "text": "T\nh",
        "tokens": [
            {
                "type": "text",
                "raw": "T\nh",
                "text": "T\nh"
            }
        ]
    },
    {
        "type": "hr",
        "raw": "---"
    }
]
```

As you can see, there's a missing spacer for the new line before the `hr` type.

Here's a demo: https://marked.js.org/demo/?outputType=lexer&text=t%0Ah%0A----&options=%7B%0A%20%22baseUrl%22%3A%20null%2C%0A%20%22breaks%22%3A%20false%2C%0A%20%22extensions%22%3A%20null%2C%0A%20%22gfm%22%3A%20true%2C%0A%20%22headerIds%22%3A%20true%2C%0A%20%22headerPrefix%22%3A%20%22%22%2C%0A%20%22highlight%22%3A%20null%2C%0A%20%22langPrefix%22%3A%20%22language-%22%2C%0A%20%22mangle%22%3A%20true%2C%0A%20%22pedantic%22%3A%20false%2C%0A%20%22sanitize%22%3A%20false%2C%0A%20%22sanitizer%22%3A%20null%2C%0A%20%22silent%22%3A%20false%2C%0A%20%22smartLists%22%3A%20false%2C%0A%20%22smartypants%22%3A%20false%2C%0A%20%22tokenizer%22%3A%20null%2C%0A%20%22walkTokens%22%3A%20null%2C%0A%20%22xhtml%22%3A%20false%0A%7D&version=master

**Expected behavior**
```json
[
    {
        "type": "paragraph",
        "raw": "T\nh",
        "text": "T\nh",
        "tokens": [
            {
                "type": "text",
                "raw": "T\nh",
                "text": "T\nh"
            }
        ]
    },
    {
        "type": "spacer",
        "raw": "\n"
    },
    {
        "type": "hr",
        "raw": "---"
    }
]
```

The issue seems to be here: https://github.com/markedjs/marked/blob/939689641817659998f63ce61ae76f0e57343268/src/Tokenizer.js#L76. 

Is there a reason we ignore spaces that are one character?

### PR DESCRIPTION
**Pull request by phillipb**
**Marked version:**
4.0.8

**Markdown flavor:** n/a

## Description
This PR stops ignoring single `\n` characters and makes sure they're properly represented in the token list for each token type in the lexer.

- Fixes #2340

## Contributor

- [x] Test(s) exist to ensure functionality and minimize regression (if no tests added, list tests covering this PR); or,

## Committer

In most cases, this should be a different person than the contributor.

- [ ] CI is green (no forced merge required).
- [ ] Squash and Merge PR following [conventional commit guidelines](https://www.conventionalcommits.org/).

### CODE REVIEW

File: test/unit/Lexer-spec.js [hunk 1]
```diff
@@ -73,8 +73,7 @@ describe('Lexer', () => {
   describe('headings', () => {
     it('depth', () => {
       expectTokens({
-        md: `
-# heading 1
+        md: `# heading 1
```
Comment (UziTech) [inline test/unit/Lexer-spec.js:76]: why do we have to remove leading newlines?

Comment (phillipb) [reply test/unit/Lexer-spec.js:76]: We don't actually have to. We could update the expect for the tests to expect the \n. This felt a little cleaner but don't have a strong preference.

Comment (UziTech) [reply test/unit/Lexer-spec.js:76]: I would rather have the tokens updated. Just keeps the tests easier to read.

**Review by UziTech — APPROVED** [review]
Thanks for working on this! :100:

**Review by styfle — APPROVED** [review]

**Pull request merged**